spec/pprof_builder_spec.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require 'cloud_profiler_agent'
+
+RSpec.describe CloudProfilerAgent::PprofBuilder do
+  subject { described_class.new(profile, start_time, end_time) }
+
+  # load_profile loads one of the example profiles created by
+  # script/generate_profile.rb
+  def load_profile(name)
+    Marshal.load(IO.binread(File.expand_path("#{name}.stackprof", __dir__)))
+  end
+
+  def get_str(index)
+    message.string_table.fetch(index)
+  end
+
+  let(:start_time) { Time.new(2020, 10, 31, 17, 12, 0) }
+  let(:end_time) { Time.new(2020, 10, 31, 17, 12, 30) }
+
+  # message is the protobuf object, which typically gets serialized and gzip'd
+  # before being sent to the Profiler API or written to disk. Rather than
+  # unzipping and deserializing all the time, we will be making a lot of
+  # assertions about the message directly.
+  let(:message) { subject.message }
+
+  context 'with :cpu profile' do
+    let(:profile) { load_profile(:cpu) }
+
+    it 'has a sample type of [["cpu", "nanoseconds"]]' do
+      expect(message.sample_type.length).to eq(1)
+
+      sample_type = message.sample_type[0]
```

Comment:
Use `.first` instead of `[0]`. There's a few places below this to also use `.first`.
```suggestion
      sample_type = message.sample_type.first
```